components/registry-facade/pkg/registry/blob.go

```diff
@@ -164,6 +172,12 @@ func (bh *blobHandler) getBlob(w http.ResponseWriter, r *http.Request) {
 
 		bh.Metrics.BlobDownloadSpeedHist.Observe(float64(n) / time.Since(t0).Seconds())
 
+		// we are returning a file from IPFS.
+		if _, ok := src.(ipfsBlobSource); ok {
+			log.WithField("digest", bh.Digest).Debug("skipping update of blob that already exists in IPFS")
+			return nil
+		}
```

Comment:
Agree. If you have any other idea to avoid adding the same thing file multiple times, open to suggestions.

Reply:
ahh, now I follow the flow. Frankly, I'd place this as an early return in the `defer`'ed function.

To avoid the type check, we could introduce a `dontCache` return value to `GetBlob` - at which point returning a struct is probably more appropriate.

Reply:
done